sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java

```diff
@@ -119,24 +124,60 @@ private void analyze(
       LOG.debug(String.format("Analyzing %s typescript file(s) with the following configuration file %s", inputFilesForThisConfig.size(), tsconfigPath));
 
       SonarTSRunnerCommand command = executableBundle.getSonarTsRunnerCommand(tsconfigPath, inputFilesForThisConfig, typeScriptRules);
-      SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
-
-      for (SonarTSRunnerResponse response : responses) {
-        FileSystem fileSystem = sensorContext.fileSystem();
-        InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
-        if (inputFile != null) {
-          saveHighlights(sensorContext, response.highlights, inputFile);
-          saveSymbols(sensorContext, response.symbols, inputFile);
-          saveMetrics(sensorContext, response, inputFile);
-          saveCpd(sensorContext, response.cpdTokens, inputFile);
-          saveFailures(sensorContext, response.issues, typeScriptRules);
-        } else {
-          LOG.error("Failed to find input file for path `" + response.filepath + "`");
+      if (checkNodeVersion(executableBundle.getNodeExecutable())) {
+        SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
+
+        for (SonarTSRunnerResponse response : responses) {
+          FileSystem fileSystem = sensorContext.fileSystem();
+          InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
+          if (inputFile != null) {
+            saveHighlights(sensorContext, response.highlights, inputFile);
+            saveSymbols(sensorContext, response.symbols, inputFile);
+            saveMetrics(sensorContext, response, inputFile);
+            saveCpd(sensorContext, response.cpdTokens, inputFile);
+            saveFailures(sensorContext, response.issues, typeScriptRules);
+          } else {
+            LOG.error("Failed to find input file for path `" + response.filepath + "`");
+          }
         }
       }
     }
   }
 
+  private boolean checkNodeVersion(String nodeExecutable) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "checkNodeVersion" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Fnode-version&issues=AWCiiE9r_Q-0dz4JAhyI&open=AWCiiE9r_Q-0dz4JAhyI)

[ISSUE_KEY]: AWCiiE9r_Q-0dz4JAhyI